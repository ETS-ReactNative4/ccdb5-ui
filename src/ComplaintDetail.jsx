import React from 'react';

export default class ComplaintDetail extends React.Component {
  render() {
    return (
      <div>
        <p>THE COMPLAINT ID YOU HAVE REQUESTED IS: { this.props.complaint_id }</p>
      </div>
    );
  }
}
