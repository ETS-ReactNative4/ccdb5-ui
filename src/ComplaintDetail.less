@import (less) "base.less";

.card-container {
  max-width: 100%;
  padding: @gutter-normal;

  nav {
    width: 100%;
    justify-content: space-between;

    a, button {
      color: @btn-bg;
      font-size: @base-font-size-px;

      &:hover {
        color: @btn-bg-hover;
      }
    }

    .a-btn__link {
      border: 0;

      span {
        margin-right: @gutter-normal;
      }
    }
  }

  h1 {
    margin: @gutter-normal 0;
  }

  h3.company-information {
    margin-top: @gutter-wide;
  }

  .card {
    display: flex;
    flex-direction: row;
    flex-wrap: nowrap;

    .card-left {
      box-sizing: border-box;
      min-width: 260px;
      width: 33%;
      padding-right: @grid_gutter-width;
    }
    .card-right {    
      width: 67%;
    }

    .cf-icon__before {
      padding-right: @gutter-narrow;
      font-size: 16px;

      &.cf-icon-approved-round {
        color: @green;
      }

      &.cf-icon-delete-round {
        color: @red;
      }

      &.cf-icon-minus-round {
        color: @gold;
      }

      &.cf-icon-help-round {
        color: @navy;
      }

      &.cf-icon-error-round {
        color: @gray-80;
      }

      &.cf-icon-clock-round {
        color: @green;
      }

      &.cf-icon-clock-round.not-timely {
        color: @gold;
      }
    }

    .layout-column {
      h3 {
        margin-top: 0;
        margin-bottom: @gutter-minimum;
      }

      h4 {
        margin-top: 0;
        margin-bottom: @gutter-minimum;
      }
    }

    .subitem {
      margin-right: @gutter-minimum;

      font-family: @webfont-demi;
    }
  }
}
