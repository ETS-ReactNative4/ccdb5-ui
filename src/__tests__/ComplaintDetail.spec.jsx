import React from 'react';
import { IntlProvider } from 'react-intl';
import renderer from 'react-test-renderer';
import ComplaintDetail from '../ComplaintDetail';

const fixture = {
  company: 'JPMORGAN CHASE & CO.',
  company_public_response: 'Company acknowledges the complaint',
  company_response: 'Closed with explanation',
  complaint_id: '2371744',
  complaint_what_happened: 'Lorem ipsum dolor sit amet',
  consumer_consent_provided: 'Consent provided',
  consumer_disputed: 'Yes',
  date_received: '2017-03-04T00:00:00',
  date_sent_to_company: '2017-03-04T00:00:00',
  has_narrative: true,
  issue: 'Account opening, closing, or management',
  product: 'Bank account or service',
  state: 'KY',
  sub_issue: 'Closing',
  sub_product: 'Checking account',
  submitted_via: 'Web',
  tags: 'Older American',
  timely: 'Yes',
  zip_code: '423XX'
}

function _buildJson(row) {
  return {
    hits: { hits: [{ _source: row }] } 
  }
}

function setupSnapshot(overrides={}) {
  const row = Object.assign({}, fixture, overrides)

  // Provide hooks for "API"
  let onSuccess, onFail

  global.fetch = jest.fn().mockImplementation((url) => {
    expect(url).toContain('@@API123456789')

    return {
      then: (x) => {
        x({ json: () => ({})})
        return {
          then: (x) => {
            onSuccess = () => x(_buildJson(row))
            return {
              catch: (y) => {onFail = y}
            }
          }
        }
      }
    }
  })

  const target = renderer.create(
    <IntlProvider locale="en">
      <ComplaintDetail complaint_id='123456789' />
    </IntlProvider>
  )

  return {target, onSuccess, onFail}
}

describe('snapshots', () => {
  it('renders without crashing', () => {
    const {target, onSuccess} = setupSnapshot()
    onSuccess()
    const tree = target.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('supports "Consumer Consent Provided" icons', () => {
    const values = [
      'Consent provided',
      'Consent not provided',
      'Consent withdrawn',
      'N/A',
      'FOO'
    ]
    
    values.forEach(v => {
      const {target, onSuccess} = setupSnapshot({consumer_consent_provided: v})
      onSuccess()
      const tree = target.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

  it('supports "Timely Response" icons', () => {
    const values = ['Yes', 'No']
    
    values.forEach(v => {
      const {target, onSuccess} = setupSnapshot({timely: v})
      onSuccess()
      const tree = target.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

  it('renders without a narrative', () => {
    const {target, onSuccess} = setupSnapshot({complaint_what_happened: ''})
    onSuccess()
    const tree = target.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders without a sub-issue', () => {
    const {target, onSuccess} = setupSnapshot({sub_issue: ''})
    onSuccess()
    const tree = target.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders without a sub-product', () => {
    const {target, onSuccess} = setupSnapshot({sub_product: ''})
    onSuccess()
    const tree = target.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders WAITING phase', () => {
    const {target} = setupSnapshot({sub_product: ''})
    const tree = target.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders ERROR phase', () => {
    const {target, onFail} = setupSnapshot({sub_product: ''})
    onFail()
    const tree = target.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
