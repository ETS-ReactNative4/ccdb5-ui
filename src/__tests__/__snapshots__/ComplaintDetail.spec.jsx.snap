exports[`initial state renders without crashing 1`] = `
<div>
  <p>
    THE COMPLAINT ID YOU HAVE REQUESTED IS: 
  </p>
</div>
`;
